swarm_dial.go

```diff
@@ -221,9 +221,6 @@ func (s *Swarm) doDial(ctx context.Context, p peer.ID) (*Conn, error) {
 		return c, nil
 	}
 
-	ctx, cancel := context.WithTimeout(ctx, DialTimeout)
```

Comment:
I'd still keep this timeout around, just make it really big (like 10 min (10x dial timeout)).

Best way to handle this imo would be to have some way tell the context to only count down when we actually do the dialing, we could then set this limit to e.g. 5 'dial-minutes'